File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.go
func topKFrequent(nums []int, k int) []int {
    hash := make(map[int]int)

    for _, v := range nums {
        if _, ok := hash[v]; !ok {
            hash[v] = 1;
        } else {
            hash[v]++;
        }
    }

    keys := make([]int, 0, 10);

    for k := range hash {
        keys = append(keys, k);
    }

    sort.Slice(keys, func (i, j int) bool {
        return hash[keys[i]] > hash[keys[j]]
    })

    ans := []int{}

    for i,v := range keys {
        if (i == k){
            break;
        }
        ans = append(ans,v);
    }


    return ans;
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        buckets = {}

        for num in nums:
            if num in buckets:
                buckets[num] += 1
            else:
                buckets[num] = 1

        heap = []

        for key,v in buckets.items():
            heapq.heappush(heap, (-v, key))

        ans = []
        while k > 0:
            v,key = heapq.heappop(heap)
            ans.append(key)
            k -= 1

        return ans